specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2023-11-01-preview/sqlServerInstances.json

```diff
@@ -0,0 +1,724 @@
+          "minimum": 0,
+          "maximum": 60,
+          "format": "int32"
+        }
+      }
+    },
+    "Monitoring": {
+      "description": "The monitoring configuration.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Indicates if monitoring is enabled for this SQL Server instance."
+        }
+      }
+    },
+    "SqlServerInstanceTelemetryRequest": {
+      "description": "The Arc SQL Server instance telemetry retrieval request.",
+      "type": "object",
+      "properties": {
+        "datasetName": {
+          "type": "string",
+          "description": "The name of the telemetry dataset to retrieve."
+        },
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The start time for the time range to fetch telemetry for. If not specified, the current time minus 1 hour is used."
+        },
+        "endTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The end time for the time range to fetch telemetry for. If not specified, the current time is used."
+        },
+        "interval": {
+          "type": "string",
+          "format": "duration",
+          "description": "The time granularity to fetch telemetry for. This is an ISO8601 duration. Examples: PT15M, PT1H, P1D",
+          "default": "PT1H"
+        },
+        "aggregationType": {
+          "type": "string",
+          "description": "The aggregation type to use for the numerical columns in the dataset.",
+          "enum": [
+            "Average",
+            "Minimum",
+            "Maximum",
+            "Sum",
+            "Count"
+          ],
+          "x-ms-enum": {
+            "name": "aggregationType",
+            "modelAsString": true
+          },
+          "default": "Average"
+        },
+        "databaseNames": {
+          "type": "array",
+          "description": "The list of database names to return telemetry for. If not specified, telemetry for all databases will be aggregated and returned.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "datasetName"
+      ]
+    },
+    "SqlServerInstanceTelemetryColumn": {
+      "description": "The telemetry column for the SQL Server instance.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the telemetry column."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the telemetry column.",
+          "enum": [
+            "bool",
+            "datetime",
+            "int",
+            "long",
+            "double",
+            "string",
+            "guid",
+            "timespan"
+          ],
+          "x-ms-enum": {
+            "name": "SqlServerInstanceTelemetryColumnType",
```

Comment:
>SqlServerInstanceTelemetryColumnType [](http://example.com/codeflow?start=21&length=36)

This name is a bit of a mouthful... for what is really just an enumerated set of data types. 
(Do you happen to support the same exact set for other operations? Any opportunity to consolidate the definitions?)